src/Microsoft.Identity.Web/AppServicesAuth/AppServicesAuthenticationTokenAcquisition.cs

```diff
@@ -87,16 +106,29 @@ private async Task<IConfidentialClientApplication> GetOrCreateApplication()
         }
 
         /// <inheritdoc/>
-        public async Task<string> GetAccessTokenForUserAsync(
+        public Task<string> GetAccessTokenForUserAsync(
             IEnumerable<string> scopes,
             string? tenantId = null,
             string? userFlow = null,
             ClaimsPrincipal? user = null,
             TokenAcquisitionOptions? tokenAcquisitionOptions = null)
         {
-            string accessToken = GetAccessToken(CurrentHttpContext?.Request.Headers);
+            var httpContext = CurrentHttpContext;
+            string accessToken;
+            if (httpContext != null)
+            {
+                // Need to lock due to https://docs.microsoft.com/en-us/aspnet/core/performance/performance-best-practices?#do-not-access-httpcontext-from-multiple-threads
+                lock (httpContext)
+                {
+                    accessToken = GetAccessToken(httpContext.Request.Headers);
+                }
+            }
+            else
```

Comment:
nit: just initialize the accessToken var with string.emtpy to avoid having this "else" clause.

Reply:
The reason it isn't initialized to empty is because doing so creates a chance that the cpu cycles to assign the variable to empty were wasted when it is overwritten. Using the else clause, we guarantee the variable is only assigned once.